### PR TITLE
Remove init method that threw runtime exception so that it causes a compile time error

### PR DESCRIPTION
**Pull request by thomaszurkan-optimizely**
Force developer to use static builderWithBlock method.

This will now throw a compile time error. Init not available:
    // Initialize an Optimizely manager
    OPTLYManagerBuilder * builder = [[OPTLYManagerBuilder alloc] init];
    builder.sdkKey = @"NEKxQMdsx5Q8byiD2hmT5J";
    self.optlyManager = [[OPTLYManager alloc] initWithBuilder:builder];

This is the only way to init properly in objective c
    self.optlyManager = [[OPTLYManager alloc] initWithBuilder:[OPTLYManagerBuilder builderWithBlock:^(OPTLYManagerBuilder * _Nullable builder) {
        builder.sdkKey = @"NEKxQMdsx5Q8byiD2hmT5J";
    }]];

### CODE REVIEW

**Comment by aliabbasrizvi** [comment]
I thought we were deprecating `init` what does  marking it as `NS_UNAVAILABLE` do?

**Comment by thomaszurkan-optimizely** [comment]
It causes a compile time error if you try and use straight [[alloc] init] which will force you to use builderWithBlock.  
I also added a logger so for instance if something goes wrong in the builder block, you can get the message. 

Before, if you made a mistake in the builder block but didn’t provide a logger, you were not notified via log message.

That said, I need to add comments

File: OptimizelySDKShared/OptimizelySDKShared/OPTLYManagerBuilder.m [hunk 1]
```diff
@@ -33,15 +33,12 @@ + (nullable instancetype)builderWithBlock:(OPTLYManagerBuilderBlock)block {
...
-    return [self initWithBlock:nil];
-}
-
 - (id)initWithBlock:(OPTLYManagerBuilderBlock)block {
-    NSParameterAssert(block);
     self = [super init];
     if (self != nil) {
-        block(self);
+        if (block != nil) {
+            block(self);
```
Comment (thomaszurkan-optimizely) [inline OptimizelySDKShared/OptimizelySDKShared/OPTLYManagerBuilder.m]: Add logging here

File: OptimizelySDKShared/OptimizelySDKShared/OPTLYManagerBuilder.h [hunk 2]
```diff
@@ -42,6 +42,7 @@ typedef void (^OPTLYManagerBuilderBlock)(OPTLYManagerBuilder * _Nullable builder
 /// User profile to be used by the client to store user-specific data.
 @property (nonatomic, readwrite, strong, nullable) id<OPTLYUserProfileService> userProfileService;
 
+- (instancetype)init NS_UNAVAILABLE;
```
Comment (thomaszurkan-optimizely) [inline OptimizelySDKShared/OptimizelySDKShared/OPTLYManagerBuilder.h]: Add comment about using builderWithBlock

**Review by mikeproeng37 — APPROVED** [review]
lgtm

**Review by aliabbasrizvi — APPROVED** [review]
Synced earlier today. Looks good.

**Pull request merged**